cmd/soci/commands/rebuild_db.go

```diff
@@ -0,0 +1,140 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package commands
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/awslabs/soci-snapshotter/fs/config"
+	"github.com/awslabs/soci-snapshotter/soci"
+	"github.com/containerd/containerd/cmd/ctr/commands"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/platforms"
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/urfave/cli"
+	"oras.land/oras-go/v2/content/oci"
+)
+
+var RebuildDBCommand = cli.Command{
+	Name:  "rebuild-db",
+	Usage: `rebuild the artifacts database. You should use this command after "rpull" so that indices/ztocs can be discovered by commands like "soci index list".`,
+	Action: func(cliContext *cli.Context) error {
```

Comment:
Strongly request the functional logic here be broken out into its own function that could be called from tests, other commands, etc. Having program logic within command functions has been a problem point with containerd in the past.

Reply:
I agree. The reason I was a little hesitant at first is that we might not eventually need this logic at all once/ if #423 happens, so I thought it might be a good idea to keep the core logic in `cmd` for now (with the exception of `RemoveOldArtifacts` since that required some explicit db logic)

I was originally thinking of having some exported func like `SyncDbWithStore` in our `soci` package how does that sound?

Reply:
That seems viable. Most of the logic here would move there.